static/app/views/alerts/list/rules/alertRulesList.tsx

```diff
@@ -0,0 +1,308 @@
+import AlertHeader from '../header';
+
+import RuleListRow from './row';
+
+type SortField = 'date_added' | 'name' | ['incident_status', 'date_triggered'];
+const defaultSort: SortField = ['incident_status', 'date_triggered'];
+
+function getAlertListQueryKey(orgSlug: string, query: Location['query']): ApiQueryKey {
+  const queryParams = {...query};
+  queryParams.expand = ['latestIncident', 'lastTriggered'];
+  queryParams.team = getTeamParams(queryParams.team!);
+
+  if (!queryParams.sort) {
+    queryParams.sort = defaultSort;
+  }
+
+  return [`/organizations/${orgSlug}/combined-rules/`, {query: queryParams}];
+}
+
+function AlertRulesList() {
+  const location = useLocation();
+  const router = useRouter();
+  const api = useApi();
+  const organization = useOrganization();
+
+  useRouteAnalyticsEventNames('alert_rules.viewed', 'Alert Rules: Viewed');
+  useRouteAnalyticsParams({
+    sort: Array.isArray(location.query.sort)
+      ? location.query.sort.join(',')
+      : location.query.sort,
+  });
+
+  const {
+    data: ruleListResponse = [],
+    refetch,
+    getResponseHeader,
+    isLoading,
+    isError,
+  } = useApiQuery<Array<CombinedMetricIssueAlerts | null>>(
+    getAlertListQueryKey(organization.slug, location.query),
+    {
+      staleTime: 0,
+    }
+  );
+
+  const handleChangeFilter = (activeFilters: string[]) => {
+    const {cursor: _cursor, page: _page, ...currentQuery} = location.query;
+    router.push({
+      pathname: location.pathname,
+      query: {
+        ...currentQuery,
+        team: activeFilters.length > 0 ? activeFilters : '',
+      },
+    });
+  };
+
+  const handleChangeSearch = (name: string) => {
+    const {cursor: _cursor, page: _page, ...currentQuery} = location.query;
+    router.push({
+      pathname: location.pathname,
+      query: {
+        ...currentQuery,
+        name,
+      },
+    });
+  };
+
+  const handleOwnerChange = (
+    projectId: string,
+    rule: CombinedMetricIssueAlerts,
+    ownerValue: string
+  ) => {
+    const endpoint =
+      rule.type === 'alert_rule'
+        ? `/organizations/${organization.slug}/alert-rules/${rule.id}`
+        : `/projects/${organization.slug}/${projectId}/rules/${rule.id}/`;
+    const updatedRule = {...rule, owner: ownerValue};
+
+    api.request(endpoint, {
+      method: 'PUT',
+      data: updatedRule,
+      success: () => {
+        addMessage(t('Updated alert rule'), 'success');
+      },
+      error: () => {
+        addMessage(t('Unable to save change'), 'error');
+      },
+    });
+  };
+
+  const handleDeleteRule = async (projectId: string, rule: CombinedMetricIssueAlerts) => {
+    try {
+      await api.requestPromise(
+        isIssueAlert(rule)
+          ? `/projects/${organization.slug}/${projectId}/rules/${rule.id}/`
+          : `/organizations/${organization.slug}/alert-rules/${rule.id}/`,
+        {
+          method: 'DELETE',
+        }
+      );
+      refetch();
```

Comment:
This works, but the delay isn't great UX. Would be better to directly update the cache:

```ts
setApiQueryData(
  queryClient,
  getAlertListQueryKey(organization.slug, location.query),
  data => data?.filter(r => r.id !== rule.id)
)
```